.github/workflows/scripts/gh-pages-prepare.sh

```diff
@@ -0,0 +1,235 @@
+#!/bin/bash
```

Comment:
This seem to complex, we should consider whether to have the docs in the docfx compatible format. The following changes would have to be made:

- rename README.md -> index.md (would loose native GitHub support when viewing the folder
- use xref when pointing to code (instead to GitHub file path)